### PR TITLE
Update components for txn file in serverless

### LINKED ISSUE #9225: Serverless trace board

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: close #9225

Problem Summary:

### What is changed and how it works?

```commit-message

```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Lloyd-Pottiger — APPROVED** [review]

File: dbms/src/Storages/KVStore/TiKVHelpers/DecodedLockCFValue.cpp [hunk 1]
```diff
@@ -81,6 +81,11 @@ inline void decodeLockCfValue(DecodedLockCFValue & res)
                 res.txn_size = readUInt64(data, len);
                 break;
             }
+            case IS_TXN_FILE_PREFIX:
+            {
+                res.is_txn_file = true;
+                break;
+            }
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/TiKVHelpers/DecodedLockCFValue.cpp:88]:
```suggestion
#if SERVERLESS_PROXY != 0
            case IS_TXN_FILE_PREFIX:
            {
                res.is_txn_file = true;
                break;
            }
#endif
```

File: dbms/src/Storages/KVStore/TiKVHelpers/DecodedLockCFValue.h [hunk 2]
```diff
@@ -39,8 +39,9 @@ struct DecodedLockCFValue : boost::noncopyable
     UInt64 min_commit_ts{0};
     std::string_view secondaries;
     std::string_view primary_lock;
+    bool is_txn_file{0};
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/TiKVHelpers/DecodedLockCFValue.h:42]:
```suggestion
    bool is_txn_file{false};
```

**Review by JaySon-Huang — COMMENTED** [review]
I think add the macro SERVERLESS_PROXY is better.

File: dbms/src/Storages/KVStore/TiKVHelpers/DecodedLockCFValue.h [hunk 3]
```diff
@@ -39,8 +39,9 @@ struct DecodedLockCFValue : boost::noncopyable
     UInt64 min_commit_ts{0};
     std::string_view secondaries;
     std::string_view primary_lock;
+    bool is_txn_file{false};
```
Comment (yongman) [inline dbms/src/Storages/KVStore/TiKVHelpers/DecodedLockCFValue.h:42]: Better to keep same format with `bool use_async_commit{0};`

Comment (CalvinNeo) [reply dbms/src/Storages/KVStore/TiKVHelpers/DecodedLockCFValue.h:42]:
```suggestion
    bool is_txn_file{0};
```

Comment (CalvinNeo) [reply dbms/src/Storages/KVStore/TiKVHelpers/DecodedLockCFValue.h]: OK,  i changed back /cc @JaySon-Huang

**Review by JaySon-Huang — APPROVED** [review]

**Pull request merged**